src/tree/revisionManagement/RevisionDraftItem.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode, Revision } from "@azure/arm-appcontainers";
+import { TreeElementBase, nonNullProp } from "@microsoft/vscode-azext-utils";
+import type { AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../utils/localize";
+import { treeUtils } from "../../utils/treeUtils";
+import type { ContainerAppModel } from "../ContainerAppItem";
+import { ScaleItem } from "../scaling/ScaleItem";
+import type { RevisionsItemModel } from "./RevisionItem";
+
+export class RevisionDraftItem implements RevisionsItemModel {
+    static readonly idSuffix: string = 'revisionDraft';
+    static readonly contextValue: string = 'revisionDraftItem';
+    static readonly contextValueRegExp: RegExp = new RegExp(RevisionDraftItem.contextValue);
+
+    id: string;
+    revisionsMode: KnownActiveRevisionsMode;
+
+    constructor(readonly subscription: AzureSubscription, readonly containerApp: ContainerAppModel, readonly revision: Revision) {
+        this.id = `${this.containerApp.id}/${RevisionDraftItem.idSuffix}`;
+        this.revisionsMode = containerApp.revisionsMode;
+    }
+
+    get revisionName(): string {
+        return nonNullProp(this.revision, 'name');
+    }
+
+    /**
+     * @example gets "rev-1" of "my-app--rev-1"
+     */
+    private get baseRevisionName(): string {
+        return this.revisionName.substring(this.revisionName.lastIndexOf('--') + 2);
```

Comment:
Another way to do this is `this.revisionName.split('--').pop()`  which I think it's a little bit cleaner.

Mostly sharing just as an FYI, not as a requested change. Both work just fine.